lms/djangoapps/program_enrollments/rest_api/v1/serializers.py

```diff
@@ -104,7 +104,11 @@ def get_curriculum_uuid(self, obj):
         return str(obj.program_enrollment.curriculum_uuid)
 
     def get_course_staff(self, obj):
-        return is_course_staff_enrollment(obj)
+        # TODO: remove is_course_staff_enrollment check once course_roles is fully impelented and data is migrated
```

Comment:
I don't this check can be entirely removed, it also looks for a [CourseAccessRoleAssignment](https://github.com/openedx/edx-platform/blob/983cde8c021acefd7a79e0a92a8b863ce1d6c754/lms/djangoapps/program_enrollments/models.py#L154) which handles enrollments for users that do not yet exist in the edx LMS. These enrollments have been uploaded by a partner prior to the program starting and users creating their accounts.

Reply:
Realizing there could be another snag in here in the fact that this API can be used by partners to assign roles. That feature would somehow need to be updated to handle the new permission system.

e.g. https://github.com/openedx/edx-platform/blob/983cde8c021acefd7a79e0a92a8b863ce1d6c754/lms/djangoapps/program_enrollments/api/writing.py#L433

Reply:
The general project plan is to first add in the appropriate permissions checks and then once the new CourseRoles logic has been fully tested begin migrating CourseAccessRole roles to CourseRoles. As a part of this migration, new roles that are assigned would be switched to assign a CourseRole instead of the current CourseAccessRole roles.

Reply:
great, I'd agree this is probably out of scope for just checking permissions and doesn't impact this PR directly. I just wanted to bring it up since it's a easily hidden featureIf we may not have accounted for.